ghcide/src/Development/IDE/Plugin/Completions/Types.hs

```diff
@@ -136,3 +138,42 @@ instance Monoid CachedCompletions where
 instance Semigroup CachedCompletions where
     CC a b c d e <> CC a' b' c' d' e' =
         CC (a<>a') (b<>b') (c<>c') (d<>d') (e<>e')
+
+data NameDetails
```

Comment:
haddock? what does this mean?

Reply:
In particular, I think it would be good to explain broadly what you're doing _somewhere_. There's quite a bit of logic spread around: you capture some information in the first pass which you want during resolution, so you store it in this datatype and put it into the `data` field of the completion item... this stuff is not obvious!